### ISSUE TITLE
TypeError PrettyPrint regression

### ISSUE THREAD

**Issue by braydonk** [goccy/go-yaml#534]
**Describe the bug**
Somewhere between v1.12.0 and v1.14.0, pretty printing for a type unmarshalling error broke.

**To Reproduce**

https://go.dev/play/p/51SbJmfrX_D

**Expected behavior**
Manually running this with `go-yaml` v1.12.0 I get:
```
[1:4] cannot unmarshal string into Go struct field a.X of type bool
    >  1 | x: hi
              ^
```
I implemented this feature in #280 

**Screenshots**

**Version Variables**
 - Go version: 1.23
 - go-yaml's Version: v1.14.0, compared against v1.12.0

**Comment by braydonk** [comment]
A bisect determined that #483 is the PR that broke this.

**Comment by braydonk** [comment]
The entire `wrapError` type was removed in that PR, which is probably why this is happening. The `FormatError` method of `TypeError` will never get called anymore, and the `Error()` method will be used on its own instead.

**Comment by goccy** [comment]
Thank you for your reports ! I've fixed this problem with `v1.14.1` !

**Comment by braydonk** [comment]
Thank you for the fix!
